src/shared/install.c

```diff
@@ -2825,28 +2849,40 @@ static int read_presets(UnitFileScope scope, const char *root_dir, Presets *pres
                         parameter = first_word(l, "enable");
                         if (parameter) {
                                 char *pattern;
+                                char **instances = NULL;
 
                                 pattern = strdup(parameter);
                                 if (!pattern)
                                         return -ENOMEM;
 
+                                r = split_pattern_into_instances(pattern, &instances);
+                                if (r < 0)
+                                        return r;
+
                                 rule = (PresetRule) {
                                         .pattern = pattern,
```

Comment:
Don't store the whole pattern here, just the unit name (and rename the field accordingly). We like to always have a "single source of truth". So there is not need to keep the original pattern, which includes the unit name and the instance names, after it has already been split.